src/bo4egenerator/duplicates.py

```diff
@@ -0,0 +1,153 @@
+"""
+Remove duplicate class and enum definitions from the generated files.
+"""
+
+import os
+import re
+from pathlib import Path
+
+
+def find_classes_and_enums_in_file(file_path: Path) -> tuple[list[str], list[str]]:
+    """
+    Find all partial class and enum definitions in a given file.
+
+    Args:
+        file_path (str): The path to the file.
+
+    Returns:
+        tuple: Two lists containing class names and enum names found in the file.
+    """
+    class_pattern = re.compile(r"\bpublic\s+partial\s+class\s+(\w+)")
+    enum_pattern = re.compile(r"\bpublic\s+enum\s+(\w+)")
+    classes = []
+    enums = []
+
+    try:
+        with open(file_path, "r", encoding="utf-8") as file:
+            content = file.read()
+            class_matches = class_pattern.findall(content)
+            enum_matches = enum_pattern.findall(content)
+            classes.extend(class_matches)
+            enums.extend(enum_matches)
+    except (PermissionError, OSError) as e:
+        print(f"Error reading file {file_path}: {e}")
```

Comment:
```suggestion
        print(f"Error reading file {file_path}: {e}")
        raise
```
oder ist es sinnvoll dann einfach fortzufahren?

ich würde mir wünschen, dass wir entweder z.b. None returnen oder eine exception werfen und den fehler an der stelle, wo der code aufgerufen wird behandeln.

Reply:
das war nur ein test brauchen es nicht überhaupt. https://github.com/Hochfrequenz/bo4e-dotnet-generator.py/pull/12/commits/84b8b9b62a1e7e9d3b97c68ce8917fccacc98a45